MatrixKit/Models/Contact/MXKContactManager.m

```diff
@@ -1230,74 +1229,125 @@ + (void)requestUserConfirmationForLocalContactsSyncWithTitle:(NSString*)title
 
 #pragma mark - Internals
 
+- (NSDictionary*)matrixContactsByMatrixIDFromMXSessions:(NSArray<MXSession*>*)mxSessions
+{
+    // The existing dictionary of contacts will be replaced by this one
+    NSMutableDictionary *matrixContactByMatrixID = [[NSMutableDictionary alloc] init];
+    for (MXSession *mxSession in mxSessions)
+    {
+        // Check all existing users
+        NSArray *mxUsers = mxSession.users;
```

Comment:
I am a bit afraid about accessing this from another thread than the main thread as there is no guarantee that the operation is thread safe.

Reply:
Yes I could make a copy of MXSession users